### PR TITLE
Implementing tiles api

### PR DESCRIPTION
**Pull request by aljordan82**
## What's this PR do?
Add a Layout grid library and implement a basic Tile.
## How should this be manually tested?
run `dky` and restart the devserver
### Front-End

- [ ] Works on IE 11 and Edge
- [ ] Works on Mobile
- [ ] Unit tests added or ticket exists to add tests

### Separation of Concerns

- [ ] Back end "business logic" is implemented in plain python, Django model classes, and Django forms. (not Django views or other frameworks)
- [ ] Concerns of UI, api, and model are in separate code units.

### Tests

- [ ] Base classes used for python tests don't do too much unnecessary work.

### Migrations

- [ ] Migrations can be reversed.
- [ ] Current production code would continue to work after this migration is applied.
- [ ] Data migrations' forward/reverse functions should be decorated with `@only_on_default`.

### CODE REVIEW

File: gulp/src/dashboard/actions/mainActions.js [hunk 1]
```diff
@@ -1,12 +1,57 @@
...
 
+export const setUserTiles = createAction('SET_USER_TILES');
+
+export function loadUserTiles() {
+  return async (dispatch, _, { api }) => {
+    try {
+      dispatch(markPageLoadingAction(true));
+      const tiles = await api.getUserTiles();
+      await dispatch(setUserTiles(tiles));
+    } catch (e) {
```
Comment (tdphillips) [inline gulp/src/dashboard/actions/mainActions.js]: Granted this is my Python experience talking, but what is this giving us that simply not catching anything doesn't?

Comment (aljordan82) [reply gulp/src/dashboard/actions/mainActions.js]: Nothing really.

**Comment by aljordan82** [comment]
DNM, I found a bug while writing unit tests

File: gulp/src/common/CommonApi.js [hunk 2]
```diff
@@ -1,7 +1,12 @@
...
 
 export default class CommonApi extends MyJobsApi {
   async getUserActivities() {
     return await this.get('/useractivities');
   }
+  async getSampleCommRecs({ limit, contact_types }) {
+    const sendUrl = '/prm/api/samplecommrecs';
+    const returnValue = await this.postJson(sendUrl, { limit, contact_types });
+    return returnValue;
+  }
```
Comment (koonsies) [inline gulp/src/common/CommonApi.js]: I'm not sure this should be in CommonApi. Do you envision other apps using this particular call?

File: gulp/src/dashboard/actions/mainActions.js (hunk 1 above)
Comment (koonsies) [reply gulp/src/dashboard/actions/mainActions.js]: I suppose it gets us the "finally".

File: gulp/src/dashboard/api.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  const formatedTiles = rawTiles.map(tile => {
+    try {
+      const newTile = {
+        ...tile,
+        dataGrid: tile.data_grid ? JSON.parse(tile.data_grid) : {},
+        tileType: tile.tile_type,
+        params: tile.params ? JSON.parse(tile.params) : {},
+      };
+      delete newTile.data_grid;
+      delete newTile.tile_type;
```
Comment (koonsies) [inline gulp/src/dashboard/api.js]: I'm not following. Why do we parse/declare these items, only to delete them out of the dictionary?

File: gulp/src/common/CommonApi.js (hunk 2 above)
Comment (aljordan82) [reply gulp/src/common/CommonApi.js]: I thought this would be used in other apps, I can put it in dashboard/api.js for now.

**Pull request merged**